code/game/gamemodes/miniantags/abduction/gland.dm

```diff
@@ -153,25 +239,71 @@
 /obj/item/organ/internal/heart/gland/spiderman
 	cooldown_low = 450
 	cooldown_high = 900
-	uses = 10
+	uses = -1
 	icon_state = "spider"
+	mind_control_uses = 2
+	mind_control_duration = 2400
 
 /obj/item/organ/internal/heart/gland/spiderman/activate()
 	to_chat(owner, "<span class='warning'>You feel something crawling in your skin.</span>")
 	owner.faction |= "spiders"
-	new /obj/structure/spider/spiderling(owner.loc)
+	var/obj/structure/spider/spiderling/S = new(owner.loc)
+	S.master_commander = owner
 
 /obj/item/organ/internal/heart/gland/egg
 	cooldown_low = 300
 	cooldown_high = 400
 	uses = -1
 	icon_state = "egg"
+	mind_control_uses = 2
+	mind_control_duration = 1800
 
 /obj/item/organ/internal/heart/gland/egg/activate()
-	to_chat(owner, "<span class='boldannounce'>You lay an egg!</span>")
-	var/obj/item/reagent_containers/food/snacks/egg/egg = new(owner.loc)
-	egg.reagents.add_reagent("sacid",20)
-	egg.desc += " It smells bad."
+	owner.visible_message("<span class='alertalien'>[owner] [pick(EGG_LAYING_MESSAGES)]</span>")
+	new /obj/item/reagent_containers/food/snacks/egg/gland(get_turf(owner))
```

Comment:
Hmm, no more sulfuric acid?

Reply:
Also, don't we normally define the list for messages used only in one place, right next to where it's used?

Reply:
These have random reagents in them now; so it could be just about any reagent.

Reply:
ohhhh, whoops